_layouts/page.html

```diff
@@ -68,6 +68,13 @@
             </div>
           {% endunless %}
         {% endif %}
+        {% if page.title == 'Glossary' %}
+          <div class="btn-group" role="group">
+          {% for letter in page.categories %}
+            <button type="button" class="btn btn-default"><a href="#{{ letter }}">{{ letter | upcase }}</a></button>
+          {% endfor %}
+          </div>
+        {% endif %}
```

Comment:
Being the glossary page is not a general purpose and hence does belong here in the general page layout file.

I guess the problem you want to solve, is that the glossary button shall appear above the headline? If so, you can do it like this:

```html
---
layout: page
title: My title
title_hidden: true
---

<!-- What needs to appear before the page title -->
<div class="btn-group" role="group">
  {% for letter in page.categories %}
    <button type="button" class="btn btn-default"><a href="#{{ letter }}">{{ letter | upcase }}</a></button>
  {% endfor %}
</div>

<!-- Insert the page title markup manually -->
<div class="page-header">
  <h1>{{ page.title }}</h1>
</div>

Foobar...
```